crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -123,309 +91,78 @@ impl Inner {
-    /// recover that `auction_id`. It also indicates whether the auction
-    /// should be indexed with its metadata. (ie. if it comes from this
-    /// environment and not from a different instance of the autopilot, e.g.
-    /// running in barn/prod). This function only returns an error
-    /// if retrying the operation makes sense.
-    async fn recover_auction_id_from_calldata(
-        ex: &mut PgConnection,
-        tx: &domain::eth::Transaction,
-        domain_separator: &model::DomainSeparator,
-    ) -> Result<AuctionIdRecoveryStatus> {
-        let tx_from = tx.from.0;
-        let settlement = match DecodedSettlement::new(&tx.input.0, domain_separator) {
-            Ok(settlement) => settlement,
-            Err(err) => {
-                tracing::warn!(
-                    ?tx,
-                    ?err,
-                    "could not decode settlement tx, unclear which auction it belongs to"
-                );
-                return Ok(AuctionIdRecoveryStatus::InvalidCalldata);
-            }
-        };
-        let auction_id = match settlement.metadata {
-            Some(bytes) => i64::from_be_bytes(bytes.0),
-            None => {
-                tracing::warn!(?tx, "could not recover the auction_id from the calldata");
-                return Ok(AuctionIdRecoveryStatus::InvalidCalldata);
-            }
-        };
-
-        let score = database::settlement_scores::fetch(ex, auction_id).await?;
-        let data_already_recorded =
-            database::settlements::already_processed(ex, auction_id).await?;
-        match (score, data_already_recorded) {
-            (None, _) => {
-                tracing::debug!(
-                    auction_id,
-                    "calldata claims to settle auction that has no competition"
-                );
-                Ok(AuctionIdRecoveryStatus::DoNotAddAuctionData(auction_id))
-            }
-            (Some(_), true) => {
-                tracing::warn!(
-                    auction_id,
-                    "settlement data already recorded for this auction"
-                );
-                Ok(AuctionIdRecoveryStatus::DoNotAddAuctionData(auction_id))
-            }
-            (Some(score), _) if score.winner.0 != tx_from.0 => {
-                tracing::warn!(
-                    auction_id,
-                    ?tx_from,
-                    winner = ?score.winner,
-                    "solution submitted by solver other than the winner"
-                );
-                Ok(AuctionIdRecoveryStatus::DoNotAddAuctionData(auction_id))
-            }
-            (Some(_), false) => Ok(AuctionIdRecoveryStatus::AddAuctionData(
-                auction_id, settlement,
-            )),
-        }
+/// Whether OnSettlementEventUpdater loop should retry on the given error.
+fn retryable(err: &settlement::Error) -> bool {
+    match err {
+        settlement::Error::Infra(_) => true,
+        settlement::Error::InconsistentData(_) => false,
```

Comment:
Should we get stuck here, wait to manually update the data to catch up later?

Reply:
No, we would get stuck on Auction not found errors which happen on prod for staging settlements.